File: src/lib/api/system.ts
import { invoke } from '@tauri-apps/api/core';

export interface OsInfo {
	platform: string;
	is_windows: boolean;
}

export function getOsInfo(): Promise<OsInfo> {
	return invoke<OsInfo>('get_os_info');
}

export function quitApp(): Promise<void> {
	return invoke('quit');
}

File: src-tauri/src/commands/system.rs
use tauri::Manager;
use crate::core::backend::BackendProcess;

#[tauri::command]
pub fn quit(app: tauri::AppHandle) {
    app.state::<BackendProcess>().kill();
    app.exit(0);
}

#[tauri::command]
pub fn restart_app(app: tauri::AppHandle) {
    tauri::process::restart(&app.env());
}

#[derive(serde::Serialize)]
pub struct OsInfo {
    pub platform: &'static str,
    pub is_windows: bool,
}

#[tauri::command]
pub fn get_os_info() -> OsInfo {
    OsInfo {
        platform: std::env::consts::OS,
        is_windows: cfg!(target_os = "windows"),
    }
}